src/zenml/models/model_models.py

```diff
@@ -13,23 +13,53 @@
 #  permissions and limitations under the License.
 """Model implementation to support Model WatchTower feature."""
 
-from typing import List, Optional, Union
+from typing import Any, Dict, List, Optional, Union
 from uuid import UUID
 
-from pydantic import BaseModel, Field
+from pydantic import BaseModel, Field, validator
 
+from zenml.model import ModelStages
+from zenml.models.artifact_models import ArtifactResponseModel
 from zenml.models.base_models import (
     WorkspaceScopedRequestModel,
     WorkspaceScopedResponseModel,
 )
 from zenml.models.constants import STR_FIELD_MAX_LENGTH, TEXT_FIELD_MAX_LENGTH
 from zenml.models.filter_models import WorkspaceScopedFilterModel
+from zenml.models.pipeline_run_models import PipelineRunResponseModel
 
 
 class ModelVersionBaseModel(BaseModel):
     """Model Version base model."""
 
-    pass
+    version: str = Field(
+        title="The name of the model version",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    description: Optional[str] = Field(
+        title="The description of the model version",
+        max_length=TEXT_FIELD_MAX_LENGTH,
+    )
+    stage: Optional[str] = Field(
+        title="The stage of the model version",
+        max_length=STR_FIELD_MAX_LENGTH,
+    )
+    _model_objects: Dict[str, UUID] = Field(
+        title="Model Objects linked to the model version",
+        default={},
+    )
+    _artifact_objects: Dict[str, UUID] = Field(
+        title="Artifacts linked to the model version",
+        default={},
+    )
+    _deployments: Dict[str, UUID] = Field(
+        title="Deployments linked to the model version",
+        default={},
+    )
+    _pipeline_runs: List[UUID] = Field(
+        title="Pipeline runs linked to the model version",
+        default=[],
+    )
```

Comment:
Yes, there are properties for users

Reply:
Then why, e.g., `_deployments` instead of `deployments`?

Reply:
Because `ModelVersionResponseModel` inherited from it holds:
```python
    @property
    def deployments(self) -> Dict[str, ArtifactResponseModel]:
        """Get all deployments linked to this version.

        Returns:
            Dictionary of Deployments as ArtifactResponseModel
        """
        return self._fetch_artifacts_from_list(self._deployments)
```
It might be worth kicking out those private folks in `ModelVersionResponseModel` directly, but apart from that, I'm not getting your concerns here.

Reply:
I'll move it to Response to avoid confusions

Reply:
maybe just rename to `self.deployment_ids` then?

Reply:
Thanks for bringing it up - caught some bugs on the way and added extra tests. Renaming is also done and I moved it out of Base to Response.

Reply:
Also added direct getters, so users can fetch specific Artifact or PipelineRun only, without populating the whole list.